source/mir/random/algorithm.d

```diff
@@ -34,868 +23,224 @@ Params:
     var = random variable (optional)
     lengths = one or more lengths
 +/
-auto randomSlice(T, G, D, size_t N)(G gen, D var, size_t[N] lengths...)
-    if (isSaturatedRandomEngine!G && isRandomVariable!D &&
+pragma(inline, false)
+auto randomSlice(G, D, size_t N)(G gen, D var, size_t[N] lengths...)
+    if (N && isSaturatedRandomEngine!G && isRandomVariable!D &&
         (is(G == class) || is(G == interface)))
 {
-    import mir.ndslice.allocation: slice;
-    import mir.ndslice.slice: slicedField;
-    return field(gen, var).slicedField(lengths).slice;
+    import mir.ndslice.allocation: uninitSlice;
+    alias T = typeof(var(gen));
+    auto ret = (()@trusted => lengths.uninitSlice!T)();
```

Comment:
You don't need a `@trusted` lambda here. `uninitSlice` uses `std.array.uninitializedArray` which is `@trusted` if `T` has no indirections.

Reply:
Same goes for the rest below. I can remove them directly though.

Reply:
Thanks!